sdcm/nemesis.py

```diff
@@ -3560,117 +3561,101 @@ def disrupt_remove_node_then_add_node(self):
 
-        node_to_remove = self.target_node
-        up_normal_nodes = self.cluster.get_nodes_up_and_normal(verification_node=node_to_remove)
+        verification_node = self.target_node
+        self.log.debug(f"Verification node is {verification_node}")
 
-        with Nemesis.run_nemesis(node_list=up_normal_nodes, nemesis_label="RemoveNodeAddNode") as verification_node:
-            self._remove_node_add_node(verification_node=verification_node, node_to_remove=node_to_remove)
+        nodes_by_rack = defaultdict(list)
+        for node in self.cluster.data_nodes:
+            if node == verification_node:
+                continue
+            self.log.debug(f"Following nodes and racks are target. Node: {node}, Rack: {node.rack}")
+            nodes_by_rack[node.rack].append(node)
 
-    def _remove_node_add_node(self, verification_node, node_to_remove, remove_node_host_id=None):
-        # node_to_remove must be different than node
-        # node_to_remove is single/last seed in cluster, before
-        # it will be terminated, choose new seed node
-        num_of_seed_nodes = len(self.cluster.seed_nodes)
-        if node_to_remove.is_seed and num_of_seed_nodes < 2:
-            new_seed_node = random.choice([n for n in self.cluster.nodes if n is not node_to_remove])
-            new_seed_node.set_seed_flag(True)
-            self.cluster.update_seed_provider()
+        eligible_racks = {rack: nodes for rack, nodes in nodes_by_rack.items() if len(nodes) >= 2}
+        random_nodes = [random.choice(nodes) for nodes in eligible_racks.values()]
 
-        # get node's host_id
-        if not remove_node_host_id:
-            removed_node_status = self.cluster.get_node_status_dictionary(
-                ip_address=node_to_remove.ip_address, verification_node=verification_node)
-            assert removed_node_status is not None, "failed to get host_id using nodetool status"
-            host_id = removed_node_status["host_id"]
-        else:
-            host_id = remove_node_host_id
+        self._remove_node_add_node(verification_node=verification_node, nodes_to_remove=random_nodes)
 
-        if SkipPerIssues('https://github.com/scylladb/scylladb/issues/21815', params=self.tester.params):
-            # TBD: To be removed after https://github.com/scylladb/scylladb/issues/21815 is resolved
-            ignore_stream_mutation_errors_due_to_issue = ignore_stream_mutation_fragments_errors
-        else:
-            ignore_stream_mutation_errors_due_to_issue = contextlib.nullcontext
+        self.log.debug("Adding nodes back started")
+        rack = max(self.cluster.racks) + 1
+        new_nodes = self._grow_cluster(rack)
+
+        try:
+            self.nodetool_cleanup_on_all_nodes_parallel()
+        finally:
+            for new_node in new_nodes:
+                self.unset_current_running_nemesis(new_node)
+
+    def _remove_node_add_node(self, verification_node, nodes_to_remove, remove_node_host_id=None):
```

Comment:
didn't find that this method add new nodes, then it could be renamed as remove nodes only

Reply:
Also i suggest to move this method from nemesis at  all and should be new operation